pkg/datapath/linux/bandwidth/ops.go

```diff
@@ -57,19 +57,30 @@ func (ops *ops) Update(ctx context.Context, txn statedb.ReadTxn, q *tables.Bandw
 	if err != nil {
 		return fmt.Errorf("QdiscList: %w", err)
 	}
-	if len(qdiscs) > 0 {
-		ok := qdiscs[0].Type() == "mq"
-		if qdiscs[0].Type() == "fq" {
-			fq, _ := qdiscs[0].(*netlink.Fq)
+	mqSeen := false
+	updatedQdiscs := 0
```

Comment:
This approach looks good. One thing this still needs is ...

egressQdiscs := len(qdiscs)

.. and then when you iterate and find clsact or ingress qdisc then just reduce egressQdiscs-- given we need to exclude them from the comparison with updatedQdiscs.

Reply:
Thanks. Updated the logic to handle for this.